hilti/runtime/include/types/tuple.h

```diff
@@ -195,12 +195,13 @@ template<typename T>
 std::string join_to_string(T&& x, const std::string& separator) {
     std::stringstream out;
 
-    std::apply(
-        [&](auto&... args) {
-            size_t i = 0;
-            ((out << rt::to_string_for_print(args) << ((i++ < std::tuple_size_v<T> - 1) ? separator : "")), ...);
-        },
-        x);
+    if constexpr ( std::tuple_size_v<T> != 0 )
+        std::apply(
+            [&](auto& arg, auto&... args) {
+                out << arg;
+                ((out << separator << args), ...);
```

Comment:
Why did this ditch the `rt::to_string_for_print` call?

Reply:
Ups, this got lost during refactoring. I added it back in.